contracts/JBMigrationOperator.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import './interfaces/IJBController.sol';
+import './interfaces/IJBDirectory.sol';
+import './interfaces/IJBMigratable.sol';
+import './interfaces/IJBPayoutRedemptionPaymentTerminal.sol';
+import './interfaces/IJBPaymentTerminal.sol';
+import './interfaces/IJBProjects.sol';
+
+
+/** 
+  @notice 
+  Allows projects to migrate their controller & terminal to 3.1 version
+  
+  @dev
+  The project owner needs to give the migration permission to this contract for carrying out the migrations for both controller & terminal.
+
+  @dev
+  The current funding cycle needs to be a reconfigured one before migration, so metadata flags allowControllerMigration, allowTerminalMigration & global.allowSetTerminals need to be set true.
+*/
+contract JBMigrationOperator {
+
+  //*********************************************************************//
+  // --------------------------- custom errors ------------------------- //
+  //*********************************************************************//
+  error UNAUTHORIZED();
+
+  //*********************************************************************//
+  // --------------- public immutable stored properties ---------------- //
+  //*********************************************************************//
+
+  /**
+    @notice
+    jbDirectory instance which keeps a track of which controller is linked to which project.
+  */
+  IJBDirectory immutable public jbDirectory;
```

Comment:
"directory" and "project" would be more consistent across this repo instead of including "jb" prefix.